### PR TITLE
Dx 1243

### PR DESCRIPTION
**Pull request by suluvai**
# Summary
Exposed contracts including IERC20 and IERC721 contracts

# Why the changes
DX-1243

# Things worth calling out

- Contracts has been moved  out of internal package so that the contracts are public and accessible

```go
// Access Registration and Core contract from the `Client` struct in `interface.go`
client.RegistrationContract
client.CoreContract

// ERC20 contract
ierc20Contract, err := client.NewIERC20Contract(context.Todo(), d.TokenAddress)
if err != nil {
return nil, err
}
_, err = ierc20Contract.Approve(opts, common.HexToAddress(client.Environment.CoreContractAddress), amount)
if err != nil {
return nil, err
}

// ERC721 contract
ierc721Contract, err := client.NewIERC721Contract(context.Todo(), d.TokenAddress)
if err != nil {
return nil, err
}
_, err = ierc721Contract.Approve(opts, common.HexToAddress(client.Environment.CoreContractAddress), tokenID)
if err != nil {
return nil, err
}
```

### CODE REVIEW

**Comment by Kastet** [comment]
> # Summary
> DX-1243 Exposed contracts including IERC20 and IERC721 contracts
> 
> # Why the changes
> # Things worth calling out

Show usage example for reviewers to understand the change.

File: imx/metadata.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func (c *Client) AddMetadataSchemaToCollection(
 	contractAddress string,
 	addMetadataSchemaToCollectionRequest api.AddMetadataSchemaToCollectionRequest,
 ) (*api.SuccessResponse, error) {
-	timestamp, signature, err := getProjectOwnerAuthorisationHeaders(l1signer)
+	timestamp, signature, err := createIMXAuthorisationHeaders(l1signer)
```
Comment (matthewmuscat) [inline imx/metadata.go:24]: I believe this is now `generateIMXAuthorisationHeaders` after the TS Core SDK changes

Comment (suluvai) [reply imx/metadata.go:24]: It is a internal function, so naming is not required to be consistent.

Comment (matthewmuscat) [reply imx/metadata.go:24]: correct but it follows the same pattern now and even internally, consistency is important for contributors.

**Review by zatlite — APPROVED** [review]

File: imx/metadata.go (hunk 1 above)
Comment (suluvai) [reply imx/metadata.go]: Not sure if I agree, but updated this time.

**Review by matthewmuscat — APPROVED** [review]

**Pull request merged**